app/helpers/newflow/login_signup_helper.rb

```diff
@@ -1,11 +1,29 @@
 module Newflow
   module LoginSignupHelper
 
-    # save (in the seession) or clear the client_app that sent the user here
+    BRI_BOOK_PARAM_NAME = :bri_book
+
+    # save (in the session) or clear the client_app that sent the user here
     def cache_client_app
       set_client_app(params[:client_id])
     end
 
+    def cache_BRI_marketing_if_present
```

Comment:
Why is this the only Salesforce field name that has all this cache clearing settings?

Reply:
Think of it as a [pre_auth_state](https://github.com/openstax/accounts/blob/master/app/controllers/legacy/signup_controller.rb#L27-L28). [Here are the methods](https://github.com/openstax/accounts/blob/a65b6c0c5cb73a0b7e86cc02948c935b4c7408ed/lib/user_session_management.rb#L107-L117) for saving and clearing the pre_auth_state cache.

Reply:
I thought there were no more preauth states?

Regardless, I'm still unclear why only this field needs to be specifically cache cleared vs other fields on the Salesforce model (which I guess you're saying is taken care of in the pre_auth_state?). For example - "Other" role name doesn't seem to do this: https://github.com/openstax/accounts/pull/923

Just trying to understand the difference.

Reply:
There is no need to cache the "Other" role name because that is submitted in a form that the user submits. In this PR, for the BRI marketing/`bri_book` field, there is a need to save in the cache (and then clear), because we get that info from a parameter in the URL and then I need I need to use it as if the user had submitted it in the form.